File: day_3/main.java
public class main {
    public static void main(String[] args) {
        int[] arr = {1, 3, 5, 7, 9, 11, 13};
        int target = 7;

        int result = binarySearch(arr, target);

        System.out.println(result);
    }

    public static int binarySearch(int[] arr, int target) {
        int l, r = 0;
        int len = arr.length - 1;

        while (l <= r) {
            int mid = (l + r) / 2;

            if (arr[mid] == target) {
                return mid;
            } else if (arr[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return -1;
    }
}

File: day_3/main.js
function binarySearch(arr, target) {
    let l = 0;
    let r = arr.length - 1;

    while (l <= r) {
        let mid = (l + r) / 2;
        if (arr[mid] === target) {
            return mid;
        } else if (arr[mid] > target) {
            l = mid + 1;
        } else {
            r = mid - 1;
        }
    }
    return -1;
}

let arr = [-1,0,3,5,9,12];
let target = 9;

console.warn(binarySearch(arr, target))